Docs: Instrument a backend plugin (#547)
grafana/plugin-tools: docusaurus/docs/introduction/backend.md

Grafana provides instance management in the backend plugin SDK to ease working with multiple configured Grafana data sources or apps, referred to as instances. This allows a plugin to simply keep state cleanly separated between instances. The SDK makes sure to optimize plugin resources by caching said instances in memory until their configuration changes in Grafana. Refer to the [HTTP Backend plugin example](https://github.com/grafana/grafana-plugin-examples/blob/main/examples/datasource-http-backend/pkg/main.go) or the [App with backend example](https://github.com/grafana/grafana-plugin-examples/blob/main/examples/app-with-backend/pkg/main.go), which shows how to use the instance management for data source and app plugins.

> Mentioned instance state is especially useful for holding client connections to downstream servers, such as HTTP, gRPC, TCP and UDP etc, to enable usage of connection pooling that optimizes usage and connection reuse to a downstream server. By using connection pooling, the plugin will not use all of the machine's available TCP connections.

Reviewer: ```suggestion
Mentioned instance state is especially useful for holding client connections to downstream servers, such as HTTP, gRPC, TCP, UDP, and so on, to enable usage of connection pooling that optimizes usage and connection reuse to a downstream server. By using connection pooling, the plugin avoids using all of the machine's available TCP connections.
```
